lib/irb/cmd/history.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require "stringio"
+require_relative "nop"
+require_relative "../pager"
+
+module IRB
+  # :stopdoc:
+
+  module ExtendCommand
+    class History < Nop
+      category "IRB"
+      description "Shows the input history. `-g [query]` or `-G [query]` allows you to filter the output."
+
+      def self.transform_args(args)
+        if match = args&.match(/\A(?<args>.+\s|)(-g|-G)\s+(?<grep>[^\s]+)\s*\n\z/)
+          args = match[:args]
+          "#{args}#{',' unless args.chomp.empty?} grep: /#{match[:grep]}/"
```

Comment:
How about `grep: #{Regexp.new(match[:grep]).inspect}` or `grep: #{match[:grep].dump}` (pass as string) so that grep regexp can include `/`

Reply:
Done! https://github.com/ruby/irb/pull/761/commits/e3bafe0655fd0fc95303373e3d1f77864a28337a